src/libraries/System.IO.FileSystem/tests/FileStream/SafeFileHandle.cs

```diff
@@ -104,14 +104,7 @@ private async Task ThrowWhenHandlePositionIsChanged(bool useAsync)
 
                     fs.WriteByte(0);
                     fsr.Position++;
-                    if (useAsync && OperatingSystem.IsWindows()) // Async I/O behaviors differ due to kernel-based implementation on Windows
-                    {
-                        Assert.Throws<IOException>(() => FSAssert.CompletesSynchronously(fs.ReadAsync(new byte[1], 0, 1)));
-                    }
-                    else
-                    {
-                        await Assert.ThrowsAsync<IOException>(() => fs.ReadAsync(new byte[1], 0, 1));
-                    }
+                    await Assert.ThrowsAsync<IOException>(() => fs.ReadAsync(new byte[1], 0, 1));
```

Comment:
Should we also have a test case for the `ReadAsync(Memory<byte>)` overload? Or is that one not affected when the position is changed?

Reply:
> Should we also have a test case for the ReadAsync(Memory<byte>) overload? Or is that one not affected when the position is changed?

We could, but our plan is to remove the position tracking at all so I don't think that it's worth it

The code that throws:

https://github.com/dotnet/runtime/blob/79ae74f5ca5c8a6fe3a48935e85bd7374959c570/src/libraries/System.Private.CoreLib/src/System/IO/LegacyFileStreamStrategy.cs#L311-L334

What we are most probably going to do soon:

https://github.com/dotnet/runtime/commit/d06f1752be351b7d5a48e2a4de79167224fe66b4